### PR TITLE
infoschemaV2: support create&drop&update table/schema for v2

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: ref #50959 

Problem Summary:

### What changed and how does it work?
1. support create & drop & update table. Need more tests for update table later.
2. support create & drop schema.
### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
image
image

- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/infoschema/builder.go [hunk 1]
```diff
@@ -577,6 +588,34 @@ func (b *Builder) applyTableUpdate(m *meta.Meta, diff *model.SchemaDiff) ([]int6
 	return tblIDs, nil
 }
 
+// TODO: more UT to check the correctness, refine the code.
```
Comment (ywqzzy) [inline pkg/infoschema/builder.go:591]:
```suggestion
// TODO: more UT to check the correctness.
```

File: pkg/infoschema/infoschema_v2.go [hunk 2]
```diff
@@ -263,6 +286,10 @@ func (is *infoschemaV2) TableByID(id int64) (val table.Table, ok bool) {
 
 	// Maybe the table is evicted? need to reload.
 	ret, err := loadTableInfo(is.r, is.Data, id, itm.dbID, is.ts, is.schemaVersion)
+
+	if ret == nil {
+		return nil, false
+	}
```
Comment (tiancaiamao) [inline pkg/infoschema/infoschema_v2.go:292]: Why not check error before ret?

**Review by tiancaiamao — APPROVED** [review]

File: pkg/infoschema/builder.go [hunk 3]
```diff
@@ -536,9 +536,17 @@ func (b *Builder) dropTableForUpdate(newTableID, oldTableID int64, dbInfo *model
...
+			} else {
+				tmpIDs = b.applyDropTable(oldDBInfo, oldTableID, tmpIDs)
+			}
 		} else {
-			tmpIDs = b.applyDropTable(dbInfo, oldTableID, tmpIDs)
+			if b.enableV2 {
+				tmpIDs = b.applyDropTableV2(diff, dbInfo, oldTableID, tmpIDs)
+			} else {
+				tmpIDs = b.applyDropTable(dbInfo, oldTableID, tmpIDs)
+			}
```
[the remark this answers is not in the document]
Comment (ywqzzy) [reply pkg/infoschema/builder.go:549]: Good idea, I will refine it.

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**